### PR TITLE
deselectAnnotation and annotationsPopUpEnabled 

### PR DESCRIPTION
**Pull request by dapetcu21**
Rebase of #386. Will test tomorrow to see if everything's ok, then merge.

### CODE REVIEW

File: ios/RCTMapboxGL/RCTMapboxGL.m [hunk 1]
```diff
@@ -35,12 +35,13 @@ @implementation RCTMapboxGL {
...
     NSURL *_styleURL;
     int _userTrackingMode;
+    RCTMGLAnnotation *_selectedAnnotation;
     BOOL _attributionButton;
     BOOL _logo;
     BOOL _compass;
     UIEdgeInsets _contentInset;
     MGLAnnotationVerticalAlignment _userLocationVerticalAlignment;
-    
+    BOOL _annotationsPopUpEnabled;
```
Comment (1ec5) [inline ios/RCTMapboxGL/RCTMapboxGL.m]: Make this a property so you don't need to implement the setter yourself.

File: ios/RCTMapboxGL/RCTMapboxGL.m [hunk 2]
```diff
@@ -470,7 +483,11 @@ - (void)mapView:(MGLMapView *)mapView didFailToLocateUserWithError:(NSError *)er
 
 -(void)mapView:(MGLMapView *)mapView didSelectAnnotation:(id<MGLAnnotation>)annotation
 {
-    if (!annotation.title || !annotation.subtitle) { return; }
+    if (!annotation.title || !annotation.subtitle) {
+        _selectedAnnotation = nil;
+        return;
+    }
+    _selectedAnnotation = annotation;
```
Comment (1ec5) [inline ios/RCTMapboxGL/RCTMapboxGL.m]: MGLMapView has a selectedAnnotations property for this.

**Review by 1ec5 — CHANGES_REQUESTED** [review]

**Comment by dapetcu21** [comment]
Tested this is working on iOS and Android. Merging.

**Comment by ericpalakovichcarr** [comment]
Thank you so much, @dapetcu21.  Any idea when this package will get updated in the npm registry?

**Comment by dapetcu21** [comment]
@bsudekum should have access to npm. Meanwhile you can `npm install --save mapbox/react-native-mapbox-gl`

**Pull request merged**